sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+                // In case of a method, we can ignore it
+                if (declSymbol is IPropertySymbol || declSymbol is IFieldSymbol)
+                {
+                    writer.WriteLine($"%{OpId(op)} = cbde.unknown : {MLIRType(id)} {GetLocation(op)} // Identifier from another assembly: {id.Identifier.ValueText}");
+                }
+                else
+                {
+                    writer.WriteLine($"// Entity from another assembly: {id.Identifier.ValueText}");
+                }
+                return;
+            }
+            var decl = declSymbol.DeclaringSyntaxReferences[0].GetSyntax();
+            if (decl == null ||                    // Not sure if we can be in this situation...
+                decl is MethodDeclarationSyntax || // We will fetch the function only when looking at the function call itself
+                decl is ClassDeclarationSyntax  || // In "Class.member", we are not interested in the "Class" part
+                decl is NamespaceDeclarationSyntax
+                )
+            {
+                // We will fetch the function only when looking at the function call itself, we just skip the identifier
+                writer.WriteLine($"// Skipped because MethodDeclarationSyntax or ClassDeclarationSyntax or NamespaceDeclarationSyntax: {id.Identifier.ValueText}");
+                return;
+            }
+
+            if (declSymbol is IFieldSymbol fieldSymbol && fieldSymbol.HasConstantValue)
+            {
+                var constValue = fieldSymbol.ConstantValue != null ? fieldSymbol.ConstantValue.ToString() : "null";
+                ExportConstant(op, fieldSymbol.Type, constValue);
+                return;
+            }
+            // IPropertySymbol could be either in a getter context (we should generate unknown) or in a setter
+            // context (we should do nothing). However, it appears that in setter context, the CFG does not have an
+            // instruction for fetching the property, so we should focus only on getter context.
+            else if (declSymbol is IFieldSymbol || declSymbol is IPropertySymbol || !AreTypesSupported(id))
+            {
+                writer.WriteLine($"%{OpId(op)} = cbde.unknown : {MLIRType(id)} {GetLocation(op)} // Not a variable of known type: {id.Identifier.ValueText}");
+                return;
+            }
+            writer.WriteLine($"%{OpId(op)} = cbde.load %{OpId(decl)} : memref<{MLIRType(id)}> {GetLocation(op)}");
+        }
+
+        private void ExtractBinaryExpression(ExpressionSyntax expr, ExpressionSyntax lhs, ExpressionSyntax rhs)
+        {
+            if (!AreTypesSupported(lhs, rhs, expr))
+            {
+                writer.WriteLine($"%{OpId(expr)} = cbde.unknown : {MLIRType(expr)} {GetLocation(expr)} // Binary expression on unsupported types {expr.Dump()}");
+                return;
+            }
+            // TODO : C#8 : Use switch expression
```

Comment:
it's possible if you rebase on master

Reply:
Unfortunately, new cases were added that no longer fall into the switch expression... So removing this outdated TODO